deepchem/data/datasets.py

```diff
@@ -1596,14 +1668,15 @@ def sanitize(elem):
 
     return iterate(self)
 
-  def transform(self,
-                transformer: "dc.trans.Transformer",
-                parallel=False,
-                **args) -> "DiskDataset":
+  # FIXME: Signature of "transform" incompatible with supertype "Dataset"
```

Comment:
It's only incompatible because you removed `**args`.  Put that back in and then you won't need the `type: ignore`.